src/js/abstract/hocs/with-styles.js

```diff
@@ -1,3 +1,7 @@
+import { add, remove } from '../../class-list';
+
+const IS_DISCONNECTED_CLASS = 'is-custom-element-disconnected';
```

Comment:
@LucaMele 
@TheDadi 
@pavestru 
We could extract this FOUC-busting approach to it's own HOC, so it's usage is more explicit.